### PR TITLE
issue: #33 - Should support undici timeout err.code

### PR DESCRIPTION
**Pull request by yohayg**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ NA] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by jkyberneees** [comment]
LGTM +1

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "fast-proxy",
-  "version": "1.6.3",
+  "version": "1.6.4",
```
Comment (mcollina) [inline package.json:3]: I prefer we bump this during release.

File: test/5.undici.test.js [hunk 2]
```diff
@@ -4,14 +4,18 @@
 const request = require('supertest')
 const bodyParser = require('body-parser')
 const expect = require('chai').expect
+
+const sleep = (milliseconds) => new Promise((resolve) => setTimeout(resolve, milliseconds))
```
Comment (mcollina) [inline test/5.undici.test.js:8]: I would use `promisify(setTimeout)` instead.

File: test/5.undici.test.js [hunk 3]
```diff
@@ -45,6 +49,10 @@ describe('undici', () => {
       res.setHeader('x-agent', 'fast-proxy')
       res.send()
     })
+    service.get('/service/timeout', async (req, res) => {
+      await sleep(2000)
```
Comment (mcollina) [inline test/5.undici.test.js:53]: why do we need 2 seconds of wait? I would prefer not to increase the slow down the testing.

File: package.json (hunk 1 above)
Comment (yohay-ma) [reply package.json]: resolved

File: test/5.undici.test.js (hunk 2 above)
Comment (yohay-ma) [reply test/5.undici.test.js]: resolved

File: test/5.undici.test.js (hunk 3 above)
Comment (yohay-ma) [reply test/5.undici.test.js]: resolved - reduced to 200ms

**Comment by yohayg** [comment]
hi @jkyberneees ,
Can you please merge this one?

**Comment by jkyberneees** [comment]
Hi @yohayg sorry for the delay on this, crazy busy on my side. Could you please resolve the conflicts here so I can merge and release both commits. 
Thanks

**Comment by yohayg** [comment]
@jkyberneees - conflicts are resolved

**Pull request merged**